### PR TITLE
Implement ember-paper

### PR DESCRIPTION
**Pull request by ultimatemonty**
WIP

### CODE REVIEW

**Comment by gschorkopf** [comment]
Taking a look at this today and tomorrow. Feel free to continue contributing on this branch if you'd like. Thanks for the work so far!

**Comment by ultimatemonty** [comment]
@gschorkopf the goal is to have everything that currently exists materialized before merging. We can work out theming later after I figure out the black magic that is ember-paper theming

**Comment by gschorkopf** [comment]
Is the addition of `package-lock.json` necessary? Why not enforce npm 6.x?

File: app/styles/app.scss [hunk 1]
```diff
@@ -1,5 +1,6 @@
-@import "ember-bootstrap/bootstrap";
+@import 'ember-paper'
+// @import "ember-bootstrap/bootstrap";
 
-@import "par-statuses";
-@import "scorecards-table";
-@import "turns-form";
+// @import "par-statuses";
```
Comment (gschorkopf) [inline app/styles/app.scss]: I always prefer removal over commented out lines. You might remove these later. I'm going commit by commit.

File: app/templates/application.hbs [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
-{{/bs-navbar}}
-
-<div class='container'>
-  {{liquid-outlet}}
+{{#paper-toolbar}}
+  {{#paper-toolbar-tools}}
+    <h2>Frolfr</h2>
+  {{/paper-toolbar-tools}}
+{{/paper-toolbar}}
+<div id="footer" style="position: fixed; left: 0; bottom: 0; width: 100%">
```
Comment (gschorkopf) [inline app/templates/application.hbs]: Prefer single quotes. We don't have an ESLint to protect against this though. However, let's move those styles to a class.

File: app/templates/application.hbs [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
+  {{#paper-toolbar}}
+    {{#paper-toolbar-tools}}
+      {{#paper-button}}{{paper-link 'Courses' 'courses'}}{{/paper-button}}
```
Comment (gschorkopf) [inline app/templates/application.hbs]: This introduces an odd style change, which kinda looks like a button within a button...

link

File: app/templates/application.hbs [hunk 4]
```diff
@@ -1,18 +1,20 @@
...
+      {{#paper-button}}{{paper-link 'Users' 'users'}}{{/paper-button}}
+      {{#paper-button}}{{paper-link 'Rounds' 'rounds.index'}}{{/paper-button}}
+    {{/paper-toolbar-tools}}
+  {{/paper-toolbar}}
 </div>
+{{#paper-content}}
+  {{liquid-outlet}}
+  <div style="right: 10px; bottom: 70px; position: fixed;">
```
Comment (gschorkopf) [inline app/templates/application.hbs]: Let's remember to move this to a class / CSS.

File: app/templates/application.hbs [hunk 5]
```diff
@@ -1,18 +1,20 @@
...
+    {{#paper-button raised=true fab=true primary=true}}{{paper-icon "play-arrow" size="36"}}{{/paper-button}}
+  </div>
+{{/paper-content}}
```
Comment (gschorkopf) [inline app/templates/application.hbs]: Prefers EOL to be an extra space. This allows git to be gentler on innocuous commits. Again, we should lint this.

File: app/templates/application.hbs [hunk 6]
```diff
@@ -1,18 +1,20 @@
...
+      {{#paper-button}}{{paper-link 'Courses' 'courses'}}{{/paper-button}}
+      {{#paper-button}}{{paper-link 'Users' 'users'}}{{/paper-button}}
+      {{#paper-button}}{{paper-link 'Rounds' 'rounds.index'}}{{/paper-button}}
+    {{/paper-toolbar-tools}}
+  {{/paper-toolbar}}
 </div>
+{{#paper-content}}
+  {{liquid-outlet}}
+  <div style="right: 10px; bottom: 70px; position: fixed;">
+    {{#paper-button raised=true fab=true primary=true}}{{paper-icon "play-arrow" size="36"}}{{/paper-button}}
```
Comment (gschorkopf) [inline app/templates/application.hbs]: Let's change this to a `plus` button and link it to `rounds.new`? Or if the user is currently playing a round, we keep it as the play button, maybe with a hint.

File: app/templates/components/filterable-resource.hbs [hunk 7]
```diff
@@ -1,7 +1,4 @@
-<div class='form-group'>
-  <label for='search'>{{if searchText searchText 'Search'}}</label>
-  {{input value=filter key-up=(action 'autoComplete' filter) class='form-control'}}
-</div>
+{{paper-input label="Search" value=filter onChange=(action 'autoComplete' filter) icon="search"}}
```
Comment (gschorkopf) [inline app/templates/components/filterable-resource.hbs]: Prefers single quotes.

File: app/routes/rounds/index.js [hunk 8]
```diff
@@ -18,9 +17,5 @@ export default ApplicationRoute.extend({
     size: {
       refreshModel: true
     }
-  },
-
-  afterModel() {
-    Ember.$('html, body').animate({ scrollTop: 0 });
```
Comment (gschorkopf) [inline app/routes/rounds/index.js:24]: Why was this removed?

File: app/components/round-resource.js [hunk 9]
```diff
@@ -1,6 +1,20 @@
 import Ember from 'ember';
 
-export default Ember.Component.extend({
-  classNames: [ 'col-xs-12', 'well', 'round-resource' ],
+const { Component, computed } = Ember;
```
Comment (gschorkopf) [inline app/components/round-resource.js:3]: 👍

Comment (gschorkopf) [reply app/components/round-resource.js:3]: Did not do this before because I am lazy 😄

File: app/components/round-resource.js [hunk 10]
```diff
@@ -1,6 +1,20 @@
...
+
+export default Component.extend({
+  classNames: [ 'round-resource' ],
   round: null,
+
+  roundIcon: computed('round.users', function() {
+    if(this.get('round.users.length') > 1) {
```
Comment (gschorkopf) [inline app/components/round-resource.js:10]: Consider onelining this: `return this.get('round.users.length') > 1 ? 'people' : 'person';`

File: app/components/round-resource.js [hunk 11]
```diff
@@ -1,6 +1,20 @@
...
+      return 'people';
+    } else {
+      return 'person';
+    }
+  }),
+
+  score: computed('users.user.id', 'scorecards.user.id', function() {
+    return this.get('scorecards').findBy('user.id', 4);
```
Comment (gschorkopf) [inline app/components/round-resource.js]: Wat. This seems bonkers to me. What is happening here?

File: app/templates/rounds/index.hbs [hunk 12]
```diff
@@ -1,12 +1,16 @@
-<h2>Rounds</h2>
+{{#paper-list}}
+  {{#each model as |round|}}
+    {{round-resource round=round}}
+  {{/each}}
+{{/paper-list}}
 
-{{#each model as |round|}}
+{{!-- {{#each model as |round|}}
```
Comment (gschorkopf) [inline app/templates/rounds/index.hbs]: Prefers not commenting things out.

File: app/templates/rounds.hbs [hunk 13]
```diff
@@ -6,4 +6,4 @@
     Friends Rounds
   {{/tabs.tab}}
 {{/paper-tabs}}
-{{liquid-outlet}}
\ No newline at end of file
+{{liquid-outlet class="md-padding"}}
```
Comment (gschorkopf) [inline app/templates/rounds.hbs]: Is this given to us by `ember-paper`? Can you specify what directing the padding/margin does in? Example: `md-padding-top`

File: app/styles/app.scss [hunk 14]
```diff
@@ -1,8 +1,42 @@
...
+$accent: 'red';
+$accentLight: #ff7961;
+$accentDark: #ba000d;
+$primaryText: #000000;
+$accentText: #fafafa;
+
 @import 'ember-paper';
 
 @import 'nav';
-// @import "ember-bootstrap/bootstrap";
```
Comment (gschorkopf) [inline app/styles/app.scss]: 👍 Sweet. Again, I'm going commit by commit, so didn't know you removed this.

File: app/templates/components/filterable-resource.hbs [hunk 15]
```diff
@@ -1,13 +1,18 @@
-{{paper-input label="Search" value=filter onChange=(action 'autoComplete' filter) icon="search"}}
-
-{{#if hasBlock}}
-  {{yield}}
-{{else}}
-  <ul>
-    {{#each filteredList as |item|}}
-      <li>
-        {{link-to item.name modelName item}}
```
Comment (gschorkopf) [inline app/templates/components/filterable-resource.hbs:12]: Removing this breaks the linkage to the appropriate model. When someone selects a user they should go to `user.scorecards`. When someone selects a course, it should go to `course.scorecards`.

File: jsconfig.json [hunk 16]
```diff
@@ -0,0 +1 @@
+{"compilerOptions":{"target":"es6","experimentalDecorators":true},"exclude":["node_modules","bower_components","tmp","vendor",".git","dist"]}
```
Comment (gschorkopf) [inline jsconfig.json]: Let's rebase this out. It's a weird commit, and even if you include it in the `.gitignore` later, the fact that it's at any point in the git history will override the `.gitignore`. See https://help.github.com/articles/removing-files-from-a-repository-s-history/

File: app/routes/application.js [hunk 17]
```diff
@@ -1,4 +1,8 @@
 import Ember from 'ember';
 import ResetScrollPositionMixin from 'frolfr-web/mixins/reset-scroll-position';
 
-export default Ember.Route.extend(ResetScrollPositionMixin, {});
+export default Ember.Route.extend(ResetScrollPositionMixin, {
+  beforeModel() {
+    this.transitionTo('rounds.index');
```
Comment (gschorkopf) [inline app/routes/application.js]: Works for me.

File: app/components/round-resource.js [hunk 18]
```diff
@@ -16,5 +16,9 @@ export default Component.extend({
 
   score: computed('users.user.id', 'scorecards.user.id', function() {
     return this.get('scorecards').findBy('user.id', 4);
-  })
+  }),
+
+  click() {
```
Comment (gschorkopf) [inline app/components/round-resource.js]: `link-to` seems pretty sufficient, right? Do we get any benefits from using `click()`?

File: app/templates/rounds/index.hbs [hunk 19]
```diff
@@ -1,21 +1,5 @@
 {{#paper-list}}
   {{#each model as |round|}}
-    {{round-resource round=round}}
+    {{round-resource round=round goToRound=(action 'goToRound')}}
```
Comment (gschorkopf) [inline app/templates/rounds/index.hbs]: https://github.com/frolfr/frolfr-web/pull/3/commits/31befd2d69fb1968e8910dabf5034c723121bfaf#diff-601bcde9e81ced6da325df4980ba5c94R16. This requires a parameter. How is this working if you're not passing in `round`?

File: jsconfig.json [hunk 20]
```diff
@@ -1 +0,0 @@
-{"compilerOptions":{"target":"es6","experimentalDecorators":true},"exclude":["node_modules","bower_components","tmp","vendor",".git","dist"]}
```
Comment (gschorkopf) [inline jsconfig.json]: 🎉

File: app/styles/variables.scss [hunk 21]
```diff
@@ -0,0 +1,12 @@
+$primary: 'blue-grey';
+$primaryLight: #c1d5e0;
+$primaryDark: #62757f;
+$accent: 'red';
+$accentLight: #ff7961;
+$accentDark: #ba000d;
+$primaryText: #000000;
+$accentText: #fafafa;
+
+$belowPar: #04B45F;
```
Comment (gschorkopf) [inline app/styles/variables.scss:10]: 💯

File: app/styles/rounds.scss [hunk 22]
```diff
@@ -0,0 +1,27 @@
+// individual round styles
```
Comment (gschorkopf) [inline app/styles/rounds.scss:1]: Prefers removal of CSS comments.

File: app/styles/rounds.scss [hunk 23]
```diff
@@ -0,0 +1,27 @@
...
+.round-resource {
+  .score {
+    font-size: 64px;
+    text-align: center;
+    border: none;
+
+    &.abovePar {
+      color: $abovePar;
```
Comment (gschorkopf) [inline app/styles/rounds.scss:9]: Nice!

File: app/components/round-resource.js [hunk 24]
```diff
@@ -14,8 +14,12 @@ export default Component.extend({
     }
   }),
 
-  score: computed('users.user.id', 'scorecards.user.id', function() {
-    return this.get('scorecards').findBy('user.id', 4);
+  score: computed('round.scorecards.[]', function() {
+    return this.get('round.scorecards.firstObject.score');//.findBy('user.id', 4);
```
Comment (gschorkopf) [inline app/components/round-resource.js]: Ah sweet. Okay I see where this is going now. That first commit had me confused.

File: app/templates/components/round-resource.hbs [hunk 25]
```diff
@@ -1,4 +1,22 @@
 {{#link-to 'round' round tagName='div'}}
+  {{#paper-card as |card|}}
+    {{#card.title as |title|}}
+      {{#title.text as |text|}}
+        {{#text.headline}}{{round.course.name}}{{/text.headline}}
+        {{#text.subhead}}{{moment-format round.createdAt 'MMMM DD, YYYY'}}{{/text.subhead}}
+      {{/title.text}}
+      {{#title.media size="md"}}
+        <div class="score par-status belowPar">-4</div>
```
Comment (gschorkopf) [inline app/templates/components/round-resource.hbs:9]: Hmm... why should we show just the "logged in user's" round score? Maybe it'd be nice to see everyone on the card's score? Or is that information overload?

**Comment by gschorkopf** [comment]
Overall, this is dope. Love the changes. There are a bunch of nitpicks I've made, but I like the direction we're going. I'm going to try to rebase some of my own work into this branch if that's cool.

**Comment by gschorkopf** [comment]
Next step: authentication?

**Comment by gschorkopf** [comment]
Working list of TODO items:

- [x] All commented out code removed
- [x] All quotes are single-quotes where appropriate
- [ ] Use paper components for: `rounds.new`, `users.new`
- [ ] Clean up footer button styles
- [x] Add extra space to EOF diffs
- [ ] Have autocomplete elements clickable -> transition to appropriate template 
- [ ] Perhaps the "play" button should go to `rounds.new`
- [ ] Add styling to `rounds` table and `turns` scorekeeping pages that look rough after removing bootstrap

**Comment by gschorkopf** [comment]
Where does this come from?

screen shot 2017-08-02 at 9 38 57 pm

I searched the repo for "Type to search" and nothing came up?

File: app/routes/index.js [hunk 26]
```diff
@@ -0,0 +1,9 @@
+import Ember from 'ember';
+
+const { Route } = Ember;
+
+export default Route.extend({
+  beforeModel() {
+    this.transitionTo('rounds.index');
```
Comment (gschorkopf) [inline app/routes/index.js:7]: Question: what's the main difference between the index route and the application route?

**Comment by gschorkopf** [comment]
Let's do the work on https://github.com/frolfr/frolfr-web/pull/4.
